ts/api/api.ts

```diff
@@ -123,6 +124,11 @@ export interface OpenYoloApi {
    * Cancels the last pending OpenYOLO request.
    */
   cancelLastOperation(): Promise<void>;
+
+  /**
+   * Sets the desired log level for the OpenYOLO library.
+   */
+  setLogLevel(level: LogLevel): Promise<void>;
```

Comment:
I wanted to be able to override the logLevel on a per request (which won't be possible without tearing down the secure channel) if I went with the above approach.

Also `OnDemandOpenYoloApi` is a terrible name (if that the real intention). I think i finally understand the differences in these interfaces. I feel like logging, and tracing is something potentially every `OpenYolo` provider should support because its a developer feature (and not a provider feature like render modes for e.g.)